### PR TITLE
COCICCS-2359:Logging enhancements - Safes added.

### PR DESCRIPTION
**Pull request by msivatmob**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java [hunk 1]
```diff
@@ -96,6 +96,12 @@
 	 * @return
 	 */
 	public ResponseEntity<String> createRole(String token, AWSLoginRole awsLoginRole, UserDetails userDetails) throws TVaultValidationException{
+		logger.debug(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
+				put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER).toString()).
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java:100]: Correct all the Code smells in logs (Remove the "toString()")

**Review by alexrajan3 — CHANGES_REQUESTED** [review]
Correct all the Code smells in logs (Remove the "toString()")

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java (hunk 1 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java]: removed toString().

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java [hunk 2]
```diff
@@ -851,13 +851,19 @@ else if (TVaultConstants.LDAP.equals(vaultAuthMethod)){
 				params.put("path",path);
 				params.put("access",access);
 				log.debug(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
-						put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER).toString()).
+						put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER)).
 						put(LogMessage.ACTION, "Add User to SDB").
-						put(LogMessage.MESSAGE, String.format ("Trying to update metadata [%s]", params.toString())).
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java:856]: toString() needed here to log the map object (params)

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java [hunk 3]
```diff
@@ -1171,11 +1181,12 @@ else if (TVaultConstants.LDAP.equals(vaultAuthMethod)){
...
 		OIDCEntityResponse oidcEntityResponse = new OIDCEntityResponse();
 		String jsonstr = JSONUtil.getJSON(safeUser);
+		String removingAccess=safeUser.getAccess();
 		log.debug(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
-			      put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER).toString()).
+			      put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER)).
 				  put(LogMessage.ACTION, "Remove User from SDB").
-			      put(LogMessage.MESSAGE, String.format ("Trying to remove user from SDB [%s]", jsonstr)).
-			      put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL).toString()).
+			      put(LogMessage.MESSAGE, String.format ("Trying to remove user from SDB [%s]]", safeUser.getPath())).
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java:1188]: Extra square bracket

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java [hunk 4]
```diff
@@ -1605,6 +1661,12 @@ else if (TVaultConstants.LDAP.equals(vaultAuthMethod)){
 	 * @return
 	 */
 	public ResponseEntity<String> addAwsRoleToSafe(String token, AWSRole awsRole) {
+		log.debug(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
+				put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER)).
+				put(LogMessage.ACTION, "Add AwsRole to SDB").
+				put(LogMessage.MESSAGE, String.format ("Trying to Add AwsRole [%s] to Safe [%s]", awsRole.getRole(),awsRole.getPath())).
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java:1667]: What will happened if awsRole object is null

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java [hunk 5]
```diff
@@ -96,6 +96,12 @@
 	 * @return
 	 */
 	public ResponseEntity<String> createRole(String token, AWSLoginRole awsLoginRole, UserDetails userDetails) throws TVaultValidationException{
+		logger.debug(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
+				put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER)).
+				put(LogMessage.ACTION, "Create AWS role").
+				put(LogMessage.MESSAGE, String.format("Trying to create AWS Role [%s]", awsLoginRole.getRole())).
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java:102]: What will happened if awsLoginRole object is null (Move the log after the request validation)

**Review by alexrajan3 — CHANGES_REQUESTED** [review]

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java (hunk 2 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java:856]: toString added.

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java (hunk 4 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java]: logs added after input validation.

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java (hunk 5 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java:106]: logs added after input validation.

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java (hunk 3 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SafesService.java]: removed.

**Review by alexrajan3 — APPROVED** [review]

**Pull request merged**